src/power_hand.c

```diff
@@ -1383,9 +1381,43 @@ TbResult magic_use_power_hand(PlayerNumber plyr_idx, MapSubtlCoord stl_x, MapSub
     return Lb_SUCCESS;
 }
 
-void stop_creatures_around_hand(char a1, unsigned short a2, unsigned short a3)
+void stop_creatures_around_hand(PlayerNumber plyr_idx, MapSubtlCoord stl_x,  MapSubtlCoord stl_y)
 {
-  _DK_stop_creatures_around_hand(a1, a2, a3);
+
+    for ( size_t i = 0; i < MID_AROUND_LENGTH; ++i )
+    {
+        struct Map* mapblk = get_map_block_at(stl_x + mid_around[i].delta_x, stl_y + mid_around[i].delta_y);
+        if(mapblk == INVALID_MAP_BLOCK)
+            continue;
+
+        unsigned long k = 0;
+        long j = get_mapwho_thing_index(mapblk);
+        while (j != 0)
+        {
+            struct Thing* thing = thing_get(j);
+            TRACE_THING(thing);
+            if (thing_is_invalid(thing))
+            {
+                ERRORLOG("Jump to invalid thing detected");
+                break;
+            }
+            j = thing->next_on_mapblk;
+            // Per thing code start
+                if ( thing_is_creature(thing) && can_thing_be_picked_up_by_player(thing, plyr_idx) && thing->owner == plyr_idx )
+                {
+                    struct CreatureControl  *cctrl = creature_control_get_from_thing(thing);
+                    cctrl->field_302 = 20;
```

Comment:
I think we can name field_302 now. This is the one place it's set I believe,... so 'wait_for_hand_turns'?